src/AppBundle/Command/ImportSaleCommand.php

```diff
@@ -83,21 +81,13 @@ private function getSales($limit, $offset)
         return $response->json();
     }
 
-    private function saveSales($jsonSales, $existingInvoices)
+    private function saveSales($jsonSales, $repository)
     {
-        foreach ($jsonSales as $jsonSale) {
-            $exists = false;
-            foreach ($existingInvoices as $existing) {
-                if (
-                    $existing['invoice'] == $jsonSale['invoice'] &&
-                    $existing['licenseId'] == $jsonSale['licenseId'] &&
-                    $existing['pluginKey'] == $jsonSale['pluginKey']
-                ) {
-                    $exists = true;
-                }
-            }
+        $newCnt = 0;
 
-            if (!$exists) {
+        foreach ($jsonSales as $jsonSale) {
+            if ($repository->findIfSaleIsNew($jsonSale['invoice'], $jsonSale['licenseId'], $jsonSale['pluginKey'])) {
```

Comment:
Hey

I've run performance tests, and it doesn't seem to matter that much, since the most time consuming operation in import is calling AM API anyway.

I've discovered another thing though — if you import all sales, then remove some of them in the DB, then run import again — they won't be reimported.

The idea is to keep our DB synced with AM data, and it might happen that some chunk of sales-data will be inserted on the AM side before the most recent sale (which is requested by default). If such event occurs, we'll have data inconsistency on our side

Do you have any thoughts or suggestions regarding this case?